layers/cmd/pipeline/pipeline.go

```diff
@@ -17,12 +17,22 @@ type PipelineContext struct {
 	*ls.Context
 	Graph       graph.Graph
 	Roots       []graph.Node
-	NextInput   func() (io.ReadCloser, error)
+	NextInput   func() (PipelineEntryInfo, io.ReadCloser, error)
 	CurrentStep int
 	Steps       []Step
 	Properties  map[string]interface{}
 	GraphOwner  *PipelineContext
-	ErrorLogger func(*PipelineContext, error) bool
+	ErrorLogger func(*PipelineContext, error)
+}
+
+type PipelineEntryInfo interface {
+	GetName() string
+}
+
+type PipelineEntryInfoFunc func() string
```

Comment:
we may extend this PipelineEntryInfo later, so instead of the func type, let's have a DefaultPipelineEntryInfo struct and use that.